internal/pkg/repository/repository.go

```diff
@@ -26,55 +26,51 @@ type Registry interface {
 
 // Repository builds and pushes images to a repository.
 type Repository struct {
-	name     string
-	registry Registry
-
-	uri string
-}
-
-// New instantiates a new Repository.
-func New(name string, registry Registry) (*Repository, error) {
-	uri, err := registry.RepositoryURI(name)
-	if err != nil {
-		return nil, fmt.Errorf("get repository URI: %w", err)
-	}
-
-	return &Repository{
-		name:     name,
-		uri:      uri,
-		registry: registry,
-	}, nil
+	Name     string
+	Registry Registry
+	Uri      string
```

Comment:
nit: 
```suggestion
	URI      string
```

Reply:
nit: I think constructors can still be useful to keep these fields private and provide different ways of initializing the object for example:
```go
type Repository struct {
  name string
  registry Registry
  uri string
}
func New(registry Registry, name string) *Repository

func NewWithURI(registry Registry, name, uri string) *Repository
```